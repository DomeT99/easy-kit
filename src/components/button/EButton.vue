<script setup lang="ts">
type ButtonType = "button" | "submit" | "reset";

interface Button {
  type?: ButtonType;
  disabled?: boolean;
  formId?: string;
  autoFocus?: boolean;
}

const props = defineProps<Button>();
</script>

<template>
  <button
    :type="props.type ?? 'button'"
    :disabled="props.disabled"
    :autofocus="props.autoFocus"
    :form="props.formId"
  >
    <slot></slot>
  </button>
</template>

<script lang="ts">
export default {
  name: "EButton",
};
</script>
