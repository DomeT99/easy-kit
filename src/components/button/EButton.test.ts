import { VueWrapper, mount } from "@vue/test-utils";
import EButton from "./EButton.vue";
import { beforeEach, describe, expect, it } from "vitest";

describe("EButton suite", () => {
  let wrapper: VueWrapper<any>;
  beforeEach(() => {
    wrapper = mount(EButton);
  });

  it("render correctly", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });
});
