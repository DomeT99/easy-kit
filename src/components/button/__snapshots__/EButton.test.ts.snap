// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EButton suite > render correctly 1`] = `"<button type=\\"button\\" class=\\"button is-primary\\" autofocus=\\"false\\"></button>"`;
